### PR TITLE
Add support for deploy.resources

### PR DESCRIPTION
**Pull request by aiordache**
This sets the task size according to the resource limits specified in the compose file(`deploy.resources.limits`).

Resource limits and reservations must be expressed in the compose file according to the compose spec(fractional numbers for cpus). 

https://github.com/compose-spec/compose-spec/blob/master/deploy.md#cpus

https://github.com/compose-spec/compose-spec/blob/master/spec.md#cpus

Fixes https://github.com/docker/ecs-plugin/issues/183

### CODE REVIEW

File: pkg/amazon/backend/convert.go [hunk 1]
```diff
@@ -111,6 +116,8 @@ func toSystemControls(sysctls types.Mapping) []ecs.TaskDefinition_SystemControl
 	return sys
 }
 
+const Mb = 1024 * 1024
```
Comment (chris-crone) [inline pkg/amazon/backend/convert.go:119]: This is a Mebibyte:

```suggestion
const MiB = 1024 * 1024
```

File: pkg/amazon/backend/cloudformation_test.go [hunk 2]
```diff
@@ -117,6 +117,44 @@ services:
...
+          memory: 8192M
+        reservations:
+          cpus: '4'
+          memory: 8192M
+`)
+	def = template.Resources["TestTaskDefinition"].(*ecs.TaskDefinition)
+	assert.Equal(t, def.Cpu, "4096")
+	assert.Equal(t, def.Memory, "8192")
+}
+
```
Comment (chris-crone) [inline pkg/amazon/backend/cloudformation_test.go:157]: Do we have a test for an invalid CPU/memory combination?

Comment (aiordache) [reply pkg/amazon/backend/cloudformation_test.go:172]: Thanks for noticing, I forgot about it. I have added one.

File: pkg/amazon/backend/cloudformation_test.go [hunk 3]
```diff
@@ -117,6 +117,59 @@ services:
...
+  test:
+    image: nginx
+    deploy:
+      resources:
+        limits:
+          cpus: '0.5'
+          memory: 2043248M
+`)
+	_, err := Backend{}.Convert(model)
+	assert.ErrorContains(t, err, "unable to find cpu/mem for the required resources")
```
Comment (nbihan-mediware) [inline pkg/amazon/backend/cloudformation_test.go:170]: I think a better error message would help here.
What about "The resources requested are not supported by ECS/Fargate"

File: pkg/amazon/backend/convert.go [hunk 4]
```diff
@@ -155,6 +162,21 @@ func toLimits(service types.ServiceConfig) (string, string, error) {
 	return "", "", fmt.Errorf("unable to find cpu/mem for the required resources")
```
Comment (nbihan-mediware) [inline pkg/amazon/backend/convert.go:162]: Ditto

File: pkg/amazon/backend/convert.go [hunk 5]
```diff
@@ -142,17 +149,32 @@ func toLimits(service types.ServiceConfig) (string, string, error) {
...
+				if limits.MemoryBytes <= m*MiB {
 					cpuLimit = strconv.FormatInt(cpu, 10)
 					memLimit = strconv.FormatInt(int64(m), 10)
 					return cpuLimit, memLimit, nil
 				}
 			}
 		}
 	}
-	return "", "", fmt.Errorf("unable to find cpu/mem for the required resources")
+	return "", "", fmt.Errorf("The resources requested are not supported by ECS/Fargate")
```
Comment (chris-crone) [inline pkg/amazon/backend/convert.go:162]: nit: errors should always start with a lower case letter.

```suggestion
	return "", "", fmt.Errorf("the resources requested are not supported by ECS on Fargate")
```

**Pull request merged**